include/jemalloc/internal/rtree_inlines.h

```diff
@@ -64,6 +64,15 @@ rtree_leafkey(uintptr_t key) {
 	return (key & mask);
 }
 
+JEMALLOC_ALWAYS_INLINE size_t
+rtree_cache_direct_map(uintptr_t key) {
```

Comment:
I discussed with @jasone a bit on hashing. Two reasons I decided to use the simple mapping here:
1) since we are talking about a few cycles of difference here on fast path, we would have to use something very fast.
2) in normal cases, we should get continuous virtual address from kernel, in which case direct mapped hash is pretty much ideal, e.g. if we have 6GB of continuous virtual address space used,  then 6 continuous entires in the rcache will work well for them, and have a smaller footprint in CPU cache (comparing to e.g. evenly distributed hash).